src/RadixDlt.NetworkGateway.GatewayApi/Validators/StateNonFungibleDetailsRequestValidator.cs

```diff
@@ -69,16 +69,25 @@
 
 namespace RadixDlt.NetworkGateway.GatewayApi.Validators;
 
-internal class NonFungibleDataRequestValidator : AbstractValidator<GatewayModel.NonFungibleDataRequest>
+internal class StateNonFungibleDetailsRequestValidator : AbstractValidator<GatewayModel.StateNonFungibleDetailsRequest>
 {
-    public NonFungibleDataRequestValidator(IOptionsSnapshot<EndpointOptions> endpointOptionsSnapshot, LedgerStateSelectorValidator ledgerStateSelectorValidator)
+    public StateNonFungibleDetailsRequestValidator(IOptionsSnapshot<EndpointOptions> endpointOptionsSnapshot, LedgerStateSelectorValidator ledgerStateSelectorValidator)
     {
-        RuleFor(x => x.Address)
+        RuleFor(x => x.ResourceAddress)
             .NotEmpty()
             .RadixAddress();
 
-        RuleFor(x => x.NonFungibleId)
-            .NotEmpty();
+        RuleFor(x => x.NonFungibleIds)
+            .NotEmpty()
+            .DependentRules(() =>
+            {
+                RuleFor(x => x.NonFungibleIds.Count)
+                    .GreaterThan(0)
+                    .LessThan(endpointOptionsSnapshot.Value.DefaultPageSize);
```

Comment:
hmm I guess that property was designed to hold default page size if value wasn't provided, not to hold upper limit. 

Guess we should use `MaxPageSize` ? I think it'd also make sense to change default value of MaxPageSize to be higher than default :D currently it's 30 and Default is 100 so if user wouldn't specify anything he'd get 100 items but he can't ask explicitly for more than 30.

Reply:
You're obviously right about both of those issues. Fixed.